src/librustc/util/ppaux.rs

```diff
@@ -802,6 +802,34 @@ impl<'tcx> fmt::Display for ty::TraitRef<'tcx> {
     }
 }
 
+impl<'tcx> ty::TypeVariants<'tcx> {
+    pub fn descr(&self) -> &'static str {
```

Comment:
That's because it's [hidden away in some error reporting logic](https://github.com/rust-lang/rust/blob/69c3d0add4a35d0a643cbdec2d91fc28808618b0/src/librustc/ty/error.rs#L186).